src/core/include/units/math.h

```diff
@@ -147,4 +147,43 @@ template<Quantity Q>
   return Q(std::numeric_limits<typename Q::rep>::epsilon());
 }
 
+/**
+ * @brief Returns the greatest quantity that is representable in Unit To that is less or equal to q.
+ *
+ * The returned value is rounded down to a quantity with unit type To
+ *
+ * @tparam q Quantity type being the base of the operation
+ * @return Quantity The rounded quantity with unit type To
+ */
+template<Unit To, typename D, typename U, typename Rep>
+[[nodiscard]] constexpr quantity<D, To, Rep> floor(const quantity<D, U, Rep>& q) noexcept
+  requires (!treat_as_floating_point<Rep>) ||
+    requires { floor(q.number()); } ||
+    requires { std::floor(q.number()); }
+{
+  auto handle_signed_results = [&q](auto&& res) constexpr
```

Comment:
```suggestion
  auto handle_signed_results = [&q](auto&& res)
```
Redundant.

Reply:
Right, I also found that lambdas are implicitly `constexpr`, but was not sure about the "best practice" and wheter being explicit improves the readability.